build_and_release.sh

```diff
@@ -18,6 +16,17 @@ platforms=(
   windows-arm64
 )
 
+if [[ "$RELEASE_ANDROID" == "true" ]]; then
+  platforms+=("android-amd64")
+  platforms+=("android-arm64")
+fi
+
+# We must know the android sdk version to build for android.
+if [[ "$RELEASE_ANDROID" == "true" && -z "$ANDROID_SDK_VERSION" ]]; then
+  echo "error: Cannot build for android without android_sdk_version." >&2
+  exit 1
+fi
```

Comment:
I like we're trying to verify the state of things, should we also test that the `ANDROID_NDK_HOME` is also a directory that exists?  Or that the `cc` is a valid directory?

Reply:
I think checking that `ANDROID_NDK_HOME` is a directory when `RELEASE_ANDROID` is `true` is reasonable 👍  

For `cc`, we arbitrarily clear it on L57 for every build target, so if we were to do validation on it, I think we'd need to do it based on the build target after `cc` is set in the loop. Do we still want to do that?

Reply:
> I think checking that ANDROID_NDK_HOME is a directory when RELEASE_ANDROID is true is reasonable 👍

I've pushed a change to check for this 🙇